nucliadb/src/nucliadb/writer/api/v1/knowledgebox.py

```
@@ -32,61 +38,101 @@
-    response_model=KnowledgeBoxObj,
     tags=["Knowledge Boxes"],
     openapi_extra={"x-hidden-operation": True},
 )
 @requires(NucliaDBRoles.MANAGER)
 @version(1)
-async def create_kb(request: Request, item: KnowledgeBoxConfig):
-    ingest = get_ingest()
-    requestpb = KnowledgeBoxNew()
-    requestpb = parse_create_kb_request(item)
-    kbobj: NewKnowledgeBoxResponse = await ingest.NewKnowledgeBox(requestpb)  # type: ignore
-    if item.slug != "":
-        slug = item.slug
+async def create_kb(request: Request, item: KnowledgeBoxConfig) -> KnowledgeBoxObj:
+    driver = get_driver()
+    rollback_learning_config = None
+
+    kbid = KnowledgeBox.new_unique_kbid()
+
+    # Onprem KBs have to call learning proxy to create it's own configuration.
+    if item.learning_configuration:
+        user_learning_config = item.learning_configuration
     else:
-        slug = kbobj.uuid
-    if kbobj.status == KnowledgeBoxResponseStatus.OK:
-        return KnowledgeBoxObj(uuid=kbobj.uuid, slug=slug)
-    elif kbobj.status == KnowledgeBoxResponseStatus.CONFLICT:
+        logger.warning(
+            "No learning configuration provided. Default will be used.",
+            extra={"kbid": kbid},
+        )
+        # learning will choose the default values
+        user_learning_config = {}
+
+    # we rely on learning to return the updated configuration with defaults and
+    # any other needed values (e.g. matryoshka settings if available)
+    learning_config = await learning_proxy.set_configuration(kbid, config=user_learning_config)
+
+    # if KB creation fails, we'll have to delete its learning config
+    async def _rollback_learning_config(kbid: str):
+        try:
+            await learning_proxy.delete_configuration(kbid)
+        except Exception:
+            logger.warning(
+                "Could not rollback learning configuration",
+                exc_info=True,
+                extra={"kbid": kbid},
+            )
+            raise
```

Do we want to raise here? This will probably make the endpoint finish with a 500 "contact your administrator"

---

We'll swallow the error. Is a warning enough for the log?